compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/replacements/StandardGraphBuilderPlugins.java

```diff
@@ -505,6 +530,10 @@ private static void registerArraysPlugins(InvocationPlugins plugins, Replacement
         r.register(new ArrayEqualsInvocationPlugin(JavaKind.Char, char[].class, char[].class));
         r.register(new ArrayEqualsInvocationPlugin(JavaKind.Int, int[].class, int[].class));
         r.register(new ArrayEqualsInvocationPlugin(JavaKind.Long, long[].class, long[].class));
+
+        r.register(new ArrayFillInvocationPlugin(JavaKind.Byte, byte[].class, byte.class));
+        r.register(new ArrayFillInvocationPlugin(JavaKind.Short, short[].class, short.class));
+        r.register(new ArrayFillInvocationPlugin(JavaKind.Int, int[].class, int.class));
```

Comment:
It would be good to support all primitive types. Is there any particular difficulty with supporting `long`? You could support `char` with the same code as `short`. You could also support `float` and `double` by using a `ReinterpretNode` to interpret the bits as integer bits and using the same code as for `int` and `long`, respectively.

Reply:
I added support for the other primitives. Testing went all good.